packages/language/src/document.js

```diff
@@ -0,0 +1,758 @@
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // sentiment = 100
+ * });
+ *
+ * //-
+ * // Verbose mode may also be enabled for more detailed results.
+ * //-
+ * var options = {
+ *   verbose: true
+ * };
+ *
+ * document.detectSentiment(options, function(err, sentiment) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // sentiment = {
+ *   //   polarity: 100,
+ *   //   magnitude: 40
+ *   // }
+ * });
+ */
+Document.prototype.detectSentiment = function(options, callback) {
+  if (is.fn(options)) {
+    callback = options;
+    options = {};
+  }
+
+  var verbose = options.verbose === true;
+
+  var grpcOpts = {
+    service: 'LanguageService',
+    method: 'analyzeSentiment'
+  };
+
+  this.request(grpcOpts, this.reqOpts, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    var originalResp = extend(true, {}, resp);
+    var sentiment = Document.formatSentiment_(resp.documentSentiment, verbose);
+
+    callback(null, sentiment, originalResp);
+  });
+};
+
+/**
+ * Take a raw response from the API and make it more user-friendly.
+ *
+ * @private
+ *
+ * @param {object} entities - A group of entities returned from the API.
+ * @param {boolean} verbose - Enable verbose mode for more detailed results.
+ * @return {object} - The formatted entity object.
+ */
+Document.formatEntities_ = function(entities, verbose) {
+  var GROUP_NAME_TO_TYPE = {
+    UNKNOWN: 'unknown',
+    PERSON: 'people',
+    LOCATION: 'places',
+    ORGANIZATION: 'organizations',
+    EVENT: 'events',
+    WORK_OF_ART: 'art',
+    CONSUMER_GOOD: 'goods',
+    OTHER: 'other'
+  };
+
+  var groupedEntities = entities.reduce(function(acc, entity) {
+    entity = extend(true, {}, entity);
+
+    var groupName = GROUP_NAME_TO_TYPE[entity.type];
+
+    entity.salience *= 100;
+
+    acc[groupName] = arrify(acc[groupName]);
+    acc[groupName].push(entity);
+    acc[groupName] = acc[groupName].sort(Document.sortByProperty_('salience'));
+
+    return acc;
+  }, {});
+
+  if (!verbose) {
+    // Simplify the response to only include an array of `name`s.
+    for (var groupName in groupedEntities) {
+      if (groupedEntities.hasOwnProperty(groupName)) {
+        groupedEntities[groupName] =
+          groupedEntities[groupName].map(prop('name'));
+      }
+    }
+  }
+
+  return groupedEntities;
+};
+
+/**
+ * Take a raw response from the API and make it more user-friendly.
+ *
+ * @private
+ *
+ * @param {object[]} sentences - A group of sentence detections returned from
+ *     the API.
+ * @param {boolean} verbose - Enable verbose mode for more detailed results.
+ * @return {object[]|string[]} - The formatted sentences or sentence descriptor
+ *     objects in verbose mode.
+ */
+Document.formatSentences_ = function(sentences, verbose) {
+  sentences = sentences.map(prop('text'));
+
+  if (!verbose) {
+    sentences = sentences.map(prop('content'));
```

Comment:
This a style nitpick, so feel free to ignore. But couldn't we just `return` here instead of re-assigning the variable?